### PR TITLE
New go definition in go.mod (v1.20)

### PR DESCRIPTION
**Pull request by iulianpascalau**
- new go definition in go.mod 
- reworked GitHub workflow files

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: golangci-lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [ main, feat/*, rc/* ]
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply .github/workflows/golangci-lint.yml:8]: agree, copy-paste error

[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply .github/workflows/golangci-lint.yml:8]: still not done

**Review by andrei-marinica — DISMISSED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,55 +1,56 @@
...
+	github.com/gin-gonic/gin v1.9.1
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/protobuf v1.5.2
 	github.com/mitchellh/mapstructure v1.4.1
-	github.com/multiversx/mx-chain-core-go v1.2.6
-	github.com/multiversx/mx-chain-crypto-go v1.2.6
-	github.com/multiversx/mx-chain-logger-go v1.0.11
-	github.com/multiversx/mx-chain-vm-common-go v1.5.0
-	github.com/multiversx/mx-components-big-int v0.1.1
+	github.com/multiversx/mx-chain-core-go v1.2.13-0.20230816090241-133da66033a9
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply go.mod]: done

File: .github/workflows/golangci-lint.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply .github/workflows/golangci-lint.yml]: fixed

**Review by AdoAdoAdo — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**